app/Http/Services/ChallengeService.php

```diff
@@ -28,9 +28,12 @@ public function store(Request $request)
         $validator = Validator::make($request->all(), [
             'events' => 'required|numeric|min:1'
         ]);
-        $challenge = new Challenge($request->all());
-        $challenge->year = Carbon::now()->year;
-        $request->get('volunteer')->challenges()->save($challenge);
+        if ($validator->passes()) {
+            $challenge = new Challenge($request->all());
+            $challenge->year = Carbon::now()->year;
+            Auth::user()->challenges()->save($challenge);
```

Comment:
Keep the If condition, just replace line 34 (new) with line 33 (old). That's because this service is used by two different authentication types so `Auth::user()` will not work for both of them. I have added `volunteer` to the request by means of a middleware, so it should fetch the authenticated user correctly.